src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Reflection/Execution/AssemblyBinderImplementation.cs

```diff
@@ -82,7 +82,7 @@ public sealed override bool Bind(RuntimeAssemblyName refName, bool cacheMissedLo
                 {
                     if (foundMatch)
                     {
-                        exception = new AmbiguousMatchException();
+                        exception = new AmbiguousMatchException(SR.Format(SR.AmbiguousMatchException_Assembly, refName.FullName));
```

Comment:
This is _similar_ to the other cases, but in the path we can directly talk about the `refName.FullName` which has everything we need to disambiguate.